wallet/src/main/java/bisq/wallet/GrpcWalletService.java

```diff
@@ -0,0 +1,248 @@
+package bisq.wallet;
+
+import bisq.common.monetary.Coin;
+import bisq.common.observable.Observable;
+import bisq.common.observable.collection.ObservableSet;
+import bisq.wallet.vo.GrpcUtxo;
+import bisq.wallet.vo.Transaction;
+import bisq.wallet.vo.TransactionInfo;
+import bisq.wallet.vo.Utxo;
+import bisq.wallets.grpc.pb.EncryptWalletRequest;
+import bisq.wallets.grpc.pb.GetBalanceRequest;
+import bisq.wallets.grpc.pb.GetSeedWordsRequest;
+import bisq.wallets.grpc.pb.GetSeedWordsResponse;
+import bisq.wallets.grpc.pb.GetUnusedAddressRequest;
+import bisq.wallets.grpc.pb.GetUnusedAddressResponse;
+import bisq.wallets.grpc.pb.GetWalletAddressesRequest;
+import bisq.wallets.grpc.pb.IsWalletEncryptedRequest;
+import bisq.wallets.grpc.pb.IsWalletEncryptedResponse;
+import bisq.wallets.grpc.pb.IsWalletReadyRequest;
+import bisq.wallets.grpc.pb.ListTransactionsRequest;
+import bisq.wallets.grpc.pb.ListUnspentRequest;
+import bisq.wallets.grpc.pb.SendToAddressRequest;
+import bisq.wallets.grpc.pb.SendToAddressResponse;
+import bisq.wallets.grpc.pb.WalletServiceGrpc;
+import com.google.common.util.concurrent.FutureCallback;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+
+public class GrpcWalletService implements WalletService {
```

Comment:
The wallet service might handle more then just grpc calls. Better extract grpc stuff to WalletGrpcClient and use that here for delegations.